server/openapi.go

```diff
@@ -51,16 +58,40 @@ func (s *Server) sqlCheckController(c echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, "Missing required SQL statement")
 	}
 
-	dbType := c.QueryParams().Get("databaseType")
-	if dbType == "" {
-		return echo.NewHTTPError(http.StatusBadRequest, "Missing required database type")
+	ctx := c.Request().Context()
+	databaseID := c.QueryParams().Get("databaseID")
+
+	var dbType db.Type
+	var catalog catalog.Catalog = &catalogService{}
+
+	if databaseID != "" {
+		id, err := strconv.Atoi(databaseID)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Invalid database ID: %v", databaseID)).SetInternal(err)
+		}
+
+		database, err := s.store.GetDatabase(ctx, &api.DatabaseFind{
+			ID: &id,
+		})
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to find database by ID: %v", databaseID)).SetInternal(err)
```

Comment:
should not be a StatusInternalServerError?

Reply:
Checked the code in the store, we will return nil if cannot find the specific database, and the error means internal error.
Code updated, keep the `StatusInternalServerError` here if err is not nil, but will return `StatusBadRequest` if the returned database is nil.